atlasdb-dbkvs/src/main/java/com/palantir/atlasdb/keyvalue/dbkvs/DbAtlasDbFactory.java

```diff
@@ -36,6 +41,14 @@ public KeyValueService createRawKeyValueService(KeyValueServiceConfig config) {
 
     @Override
     public TimestampService createTimestampService(KeyValueService rawKvs) {
-        throw new UnsupportedOperationException("Cannot instantiate a TimestampService from a relational key value service.");
+        Preconditions.checkArgument(rawKvs instanceof DbKvs, "DbAtlasDbFactory expects a raw kvs of type DbKvs, found %s", rawKvs.getClass());
+        DbKvs dbkvs = (DbKvs) rawKvs;
+        return PersistentTimestampService.create(
+                new InDbTimestampBoundStore(
+                        new HikariCPConnectionManager(
```

Comment:
Is the intention to have this then shared with the raw kvs?  I know we've discussed just want to get it decided since this PR doesn't yet include the raw kvs creation.